### PR TITLE
add deprecatiion warning to relevancy_index

### PR DESCRIPTION
**Pull request by bardkw**
Purpose of this PR is to add a deprecation warning to the RelevancyIndex interface in favor of RankRelevancy

### CODE REVIEW

File: docs/release_notes/pending_release.rst [hunk 1]
```diff
@@ -4,5 +4,9 @@ Pending Release Notes
 Updates / New Features
 ----------------------
 
+Interfaces
+
+* Deprecated the RelevancyIndex  interface in favor of RankRelevancy
```
Comment (Purg) [inline docs/release_notes/pending_release.rst]: Seem to be two spaces here between words for some reason.
```suggestion
* Deprecated the RelevancyIndex interface in favor of RankRelevancy
```

**Review by Purg — COMMENTED** [review]
Is the intention for the warning to only be emitted when the module is imported? I ask due to the warning being in the class level vs. the module level.

We should also report our strategy for removal in the message emitted. I say that but have not declared anywhere that we have an official "strategy". One example strategy to borrow from might be Numpy ([NEP 23](https://numpy.org/neps/nep-0023-backwards-compatibility.html)). We should probably start a doc page (maybe in smqtk-core for general applicability) with whatever we decide on.

I was thinking maybe at least just a warning that a "future release may remove this functionality" implying the next release, v0.16.0, will still have this class in it so that there will at least be one release to allow for visibility of the warning. I like the minimum real-world timing aspect in the numpy strategy, e.g. at least retained for N months (6 in their case).

**Comment by bardkw** [comment]
I thought it would make sense if it was class level, but i see what you're saying and will put it at module level.
Would adding it at module level mean putting the warning just outside the class in the file?

**Comment by Purg** [comment]
Yeah, I think this is fine. I don't know of a better way to do this in the current context.

**Review by Purg — APPROVED** [review]

**Pull request merged**